keras/src/backend/openvino/numpy.py

```diff
@@ -1030,9 +1032,28 @@ def moveaxis(x, source, destination):
 
 
 def nan_to_num(x, nan=0.0, posinf=None, neginf=None):
-    raise NotImplementedError(
-        "`nan_to_num` is not supported with openvino backend"
-    )
+    x = get_ov_output(x)
+    dtype = x.get_element_type()
+    nan_val = ov_opset.constant(nan, dtype).output(0)
+    posinf_val = ov_opset.constant(
+        posinf if posinf is not None else DTYPES_MAX[dtype], dtype
+    ).output(0)
+    neginf_val = ov_opset.constant(
+        neginf if neginf is not None else DTYPES_MIN[dtype], dtype
+    ).output(0)
+    posinf_mask = ov_opset.is_inf(
+        ov_opset.convert(x, Type.f32),
```

Comment:
probably it is a bug on OpenVINO side. Please create issue for non-f32 types in OpenVINO repo. For now, this WA absolutely works for us

Reply:
Done , I have created the issue.
Thank you